sim/warlock/talents_affliction.go

```diff
@@ -130,43 +138,37 @@ func (warlock *Warlock) registerShadowEmbrace() {
 
 	warlock.ShadowEmbraceAuras = warlock.NewEnemyAuraArray(warlock.ShadowEmbraceDebuffAura)
 
-	warlock.RegisterAura(core.Aura{
-		Label:    "Shadow Embrace Talent Hidden Aura",
-		Duration: core.NeverExpires,
-		OnReset: func(aura *core.Aura, sim *core.Simulation) {
-			aura.Activate(sim)
-		},
-		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
-			if (spell == warlock.ShadowBolt || spell == warlock.Haunt) && result.Landed() {
-				aura := warlock.ShadowEmbraceAuras.Get(result.Target)
-				aura.Activate(sim)
-				aura.AddStack(sim)
-			}
-		},
-	})
+	core.MakePermanent(
+		warlock.RegisterAura(core.Aura{
+			Label: "Shadow Embrace Talent Hidden Aura",
+			OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if (spell == warlock.ShadowBolt || spell == warlock.Haunt) && result.Landed() {
+					aura := warlock.ShadowEmbraceAuras.Get(result.Target)
+					aura.Activate(sim)
+					aura.AddStack(sim)
+				}
+			},
+		}))
 }
 
 func (warlock *Warlock) registerEverlastingAffliction() {
 	procChance := []float64{0, 0.33, 0.66, 1.0}[warlock.Talents.EverlastingAffliction]
 
-	warlock.RegisterAura(core.Aura{
-		Label:    "EverlastingAffliction Talent",
-		Duration: core.NeverExpires,
-		OnReset: func(aura *core.Aura, sim *core.Simulation) {
-			aura.Activate(sim)
-		},
-		OnPeriodicDamageDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
-			//TODO: Does this apply on haunt damage or just the dot?
-			if spell != warlock.DrainSoul || spell != warlock.Haunt {
-				return
-			}
+	core.MakePermanent(
+		warlock.RegisterAura(core.Aura{
+			Label: "EverlastingAffliction Talent",
+			OnPeriodicDamageDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				//TODO: Does this apply on haunt damage or just the dot?
+				if spell != warlock.DrainSoul || spell != warlock.Haunt {
+					return
+				}
 
-			if warlock.Corruption.Dot(aura.Unit).IsActive() && sim.Proc(procChance, "EverlastingAffliction") {
-				//TODO: Should this Rollover or Apply like other dots in cata?
-				warlock.Corruption.Dot(aura.Unit).Rollover(sim)
-			}
-		},
-	})
+				if warlock.Corruption.Dot(aura.Unit).IsActive() && sim.Proc(procChance, "EverlastingAffliction") {
+					//TODO: Should this Rollover or Apply like other dots in cata?
+					warlock.Corruption.Dot(aura.Unit).Rollover(sim)
```

Comment:
I think in cata everything is an Apply now. The new Apply does rollover if the spell is active (as in it updates duration and values but keeps the tick timer)

Reply:
I wasn't sure if one method (rollover) vs the other (apply) brought the snapshot bonuses along with it or resnapshotted in cata is that a thing still? I know there are some changes.

Reply:
I referred back to this article which states that the roll over mechanic which happened in wrath changed so that buffs will re-snapshot on each apply/refresh. https://cynwise.wordpress.com/2011/11/18/how-warlock-dots-work-in-cataclysm-part-ii-dot-refreshing-proc-stacking-and-getting-your-warlock-black-belt/
So I will update these all to apply.